packages/flutter_tools/templates/package_ffi/cpp/hook/build.dart.tmpl

```diff
@@ -0,0 +1,26 @@
+import 'package:native_toolchain_c/native_toolchain_c.dart';
+import 'package:logging/logging.dart';
+import 'package:native_assets_cli/native_assets_cli.dart';
+
+const packageName = '{{projectName}}';
+
+void main(List<String> args) async {
+  await build(args, (config, output) async {
```

Comment:
> @stuartmorgan Do you know if we have Rust and Go toolchains on our CI?

I don't know if they are in the base images we use for LUCI, but I would guess not. If not, then not currently since I don't see any CIPD packages in the CI config for either of them. It's possible that packages exist that we could just add, but I don't know offhand.

Reply:
@Buchimi this requires some experimentation. You can try invoking the compilers as command-line tools in the tests. (After you've signed the CLA and the tests run.)

>  It's possible that packages exist that we could just add, but I don't know offhand.

Just browsing around on cipd:

```
dacoharkes-macbookpro2:~ dacoharkes$ cipd ls <f...>/go/
<f...>/go/linux-amd64
<f...>/go/linux-arm64
<f...>/go/mac-amd64
<f...>/go/mac-arm64
<f...>/go/windows-amd64
dacoharkes-macbookpro2:~ dacoharkes$ cipd ls <f...>/third_party/go
<f...>/third_party/go/linux-amd64
<f...>/third_party/go/linux-arm64
<f...>/third_party/go/mac-amd64
dacoharkes-macbookpro2:~ dacoharkes$ cipd ls <f...>/third_party/rust
<f...>/third_party/rust/host/
<f...>/third_party/rust/linux-amd64
<f...>/third_party/rust/linux-arm64
<f...>/third_party/rust/mac-amd64
<f...>/third_party/rust/target/
<f...>/third_party/rust/windows-amd64
dacoharkes-macbookpro2:~ dacoharkes$ cipd ls <f...>/third_party/rust_bindgen
<f...>/third_party/rust_bindgen/linux-amd64
<f...>/third_party/rust_bindgen/mac-amd64
<f...>/third_party/rust_bindgen/mac-arm64
```

So we seem to have them.

@stuartmorgan can you point me to where our LUCI bots for flutter/flutter are configured? I only see a DEPS file in flutter/engine. https://github.com/flutter/engine/blob/main/DEPS

(If we can't make it to work on the LUCI bots here on the Flutter CI, our aim should be to do the same that we did for FFIgen: Add some script on that repo that runs `flutter create --template=package_ffi --native-language=...` and runs FFIgen and cbindgen/cgo. https://github.com/dart-archive/ffigen/pull/433/files In that case, we'd need to merge this PR first so that the template is available on the flutter master channel.)